buildchain/buildchain/salt_tree.py

```diff
@@ -81,158 +81,147 @@ def task__deploy_salt_tree() -> Iterator[types.TaskDict]:
+
+    Path('salt/metalk8s/kubernetes/CNI/calico/configured.sls'),
+    Path('salt/metalk8s/kubernetes/CNI/calico/deployed.sls'),
+    Path('salt/metalk8s/kubernetes/CNI/calico/init.sls'),
+    Path('salt/metalk8s/kubernetes/CNI/calico/installed.sls'),
+
+    Path('salt/metalk8s/kubernetes/controller-manager/init.sls'),
+    Path('salt/metalk8s/kubernetes/controller-manager/installed.sls'),
+    Path('salt/metalk8s/kubernetes/controller-manager/kubeconfig.sls'),
+
+    Path('salt/metalk8s/kubernetes/coredns/deployed.sls'),
+    Path('salt/metalk8s/kubernetes/coredns/files/coredns-deployment.yaml.j2'),
+
+    Path('salt/metalk8s/kubernetes/etcd/certs/healthcheck-client.sls'),
+    Path('salt/metalk8s/kubernetes/etcd/certs/init.sls'),
+    Path('salt/metalk8s/kubernetes/etcd/certs/peer.sls'),
+    Path('salt/metalk8s/kubernetes/etcd/certs/server.sls'),
+    Path('salt/metalk8s/kubernetes/etcd/files/manifest.yaml'),
+    Path('salt/metalk8s/kubernetes/etcd/init.sls'),
+    Path('salt/metalk8s/kubernetes/etcd/installed.sls'),
+
+    Path('salt/metalk8s/kubernetes/kubelet/configured.sls'),
+    Path('salt/metalk8s/kubernetes/kubelet/files/kubeadm.env'),
     # pylint:disable=line-too-long
-    Path('salt/metalk8s/kubeadm/init/kubelet-start/files/service-kubelet-systemd.conf'),
-    Path('salt/metalk8s/kubeadm/init/kubelet-start/init.sls'),
-
-    Path('salt/metalk8s/kubeadm/init/preflight/init.sls'),
-    Path('salt/metalk8s/kubeadm/init/preflight/mandatory.sls'),
-    Path('salt/metalk8s/kubeadm/init/preflight/recommended.sls'),
-
-    Path('salt/metalk8s/kubeadm/init/kubeconfig/admin.sls'),
-    Path('salt/metalk8s/kubeadm/init/kubeconfig/controller-manager.sls'),
-    Path('salt/metalk8s/kubeadm/init/kubeconfig/files/service-kubelet-systemd.conf'),
-    Path('salt/metalk8s/kubeadm/init/kubeconfig/init.sls'),
-    Path('salt/metalk8s/kubeadm/init/kubeconfig/kubelet.sls'),
-    Path('salt/metalk8s/kubeadm/init/kubeconfig/lib.sls'),
-    Path('salt/metalk8s/kubeadm/init/kubeconfig/scheduler.sls'),
-
-    Path('salt/metalk8s/kubeadm/init/mark-control-plane/init.sls'),
+    Path('salt/metalk8s/kubernetes/kubelet/files/service-standalone-systemd.conf'),
+    Path('salt/metalk8s/kubernetes/kubelet/files/service-systemd.conf'),
+    Path('salt/metalk8s/kubernetes/kubelet/init.sls'),
+    Path('salt/metalk8s/kubernetes/kubelet/installed.sls'),
+    Path('salt/metalk8s/kubernetes/kubelet/running.sls'),
+    Path('salt/metalk8s/kubernetes/kubelet/standalone.sls'),
+
+    Path('salt/metalk8s/kubernetes/kube-proxy/deployed.sls'),
+
     targets.TemplateFile(
         task_name='configured.sls',
-        source=constants.ROOT/'salt'/'metalk8s'/'kubeadm'/'init'/'mark-control-plane'/'configured.sls.in',
-        destination=constants.ISO_ROOT/'salt'/'metalk8s'/'kubeadm'/'init'/'mark-control-plane'/'configured.sls',
+        # pylint:disable=line-too-long
+        source=constants.ROOT/'salt'/'metalk8s'/'kubernetes'/'mark-control-plane'/'deployed.sls.in',
```

Comment:
Using `joinpath` may be clearer here, since you could split the line.

e.g:

```python
         source=constants.ROOT.joinpath('salt', 'metalk8s', 'kubernetes',
                                        'mark-control-plane', 'deployed.sls.in')
```

ditto below.

Reply:
As a side note, I tried passing `black` on this file, and it doesn't look super nice:

```python
    targets.TemplateFile(
        task_name="configured.sls",
        source=constants.ROOT
        / "salt"
        / "metalk8s"
        / "kubernetes"
        / "mark-control-plane"
        / "deployed.sls.in",
        destination=constants.ISO_ROOT
        / "salt"
        / "metalk8s"
        / "kubernetes"
        / "mark-control-plane"
        / "deployed.sls",
        context={"VERSION": constants.SHORT_VERSION},
        file_dep=[constants.VERSION_FILE],
    ),
```

Reply:
Maybe it's better with `joinpath`?

Anyway, the point of `black` is not having something nice: it may produce something ugly that everybody hates: but at least it's automated and there is no more debate in review AND the code layout is consistent)

On QuadIron, where we use `clang-format`, the layout is weird and ugly on some point as well.

Reply:
I'll take readable/maintainable code that may not be 100% consistent across the board over an unreadable mess generated by some tool that is 100% consistent with... 'something'.

At least I've never known `clang-format` turn reasonable source like the original above into a complete mess like what this tool generates.

Also, if your formatter code requires you *to make code changes to make thing look 'somewhat' ok*, then something's very wrong.


Seriously, there are more important things to tackle than making sure that every single character lands into exactly this or that position according to some arbitrary rule. As long as the code is

- Readable
- Somewhat consistent (e.g. don't mix snake-case with camel-case or whatever all over the place)
- Has some reasonable automated checks for the above in place

then why would one need to go beyond that.

Reply:
The code was bad to begin with (line too long), so it needs to be rewritten anyway (then, if with `joinpath` it's still a mess then yeah this deserve a bug report for `black`).